### ISSUE TITLE
Optimize Table Segment Read/Write paths

### ISSUE THREAD

**Issue by andreipaduroiu** [pravega/pravega#4788]
**Problem description**
Things to consider:
1. Zero-copy approach. Use the ByteBufs from the Wire protocol and compose them all the way to the BK write without having to copy. Consider changing WireCommands, PravegaRequestProcessor, EntrySerializer.
2. WriterTableProcessor. Attempt to reduce copies to zero.
3. After #4763 is merged in, we need to figure out a way to not copy the entire key just because we want to prepend the "E"/"I" prefix. Consider moving away from `ArrayView`s to `BufferView`s . This will likely require more extensive code changes as many places (including the hasher) expects some byte array to work with.

**Problem location**
Segment store.

**Suggestions for an improvement**
See above.

**Comment by andreipaduroiu** [comment]
Superseded by #4808.
